module/VuFind/tests/integration-tests/src/VuFindTest/Auth/ShibbolethWithWAYFTest.php

```diff
@@ -0,0 +1,237 @@
+    {
+        if (null === $config) {
+            $config = $this->getAuthConfig($proxy);
+        }
+        if (null === $shibConfig) {
+            $shibConfig = $this->getShibbolethConfig();
+        }
+        $obj = new ShibbolethWithWAYF(
+            $this->createMock(\Laminas\Session\ManagerInterface::class),
+            $shibConfig
+        );
+        $initializer = new \VuFind\ServiceManager\ServiceInitializer();
+        $initializer($this->getServiceManager(), $obj);
+        $obj->setConfig($config);
+        return $obj;
+    }
+
+    /**
+     * Get a working configuration for the LDAP object
+     *
+     * @param bool   $proxy  shibboleth is behind HTTP proxy
+     *
+     * @return Config
+     */
+    public function getAuthConfig($proxy = false)
+    {
+        $config = new Config(
+            [
+                'login' => 'http://localhost/Shibboleth.sso/Login',
+                'username' => 'username',
+                'email' => 'email',
+                'cat_username' => 'userLibraryId',
+                'proxy' => $proxy,
+            ], true
+        );
+        return new Config(['Shibboleth' => $config], true);
+    }
+
+    /**
+     * Get a working configuration for the Shibboleth object
+     *
+     * @return Config
+     */
+    public function getShibbolethConfig()
+    {
+        $example1 = new Config(
+            [
+                'entityId' => 'https://idp1.example.org/',
+                'username' => 'username',
+                'email' => 'email',
+                'cat_username' => 'userLibraryId',
+            ], true
+        );
+        $example2 = new Config(
+            [
+                'entityId' => 'https://idp2.example.org/',
+                'username' => 'eppn',
+                'email' => 'email',
+                'cat_username' => 'alephId',
+                'userattribute_1' => 'eduPersonScopedAffiliation',
+                'userattribute_value_1' => 'member@example.org',
+            ], true
+        );
+        $config = [
+            'example1' => $example1,
+            'example2' => $example2,
+        ];
+        return new Config($config, true);
+    }
+
+    /**
+     * Support method -- get parameters to log into an account (but allow override of
+     * individual parameters so we can test different scenarios).
+     *
+     * @param array $overrides Associative array of parameters to override.
+     *
+     * @return \Laminas\Http\Request
+     */
+    protected function getLoginRequest($user, $proxy)
+    {
+        $request = new \Laminas\Http\PhpEnvironment\Request();
+        if ($proxy) {
+            $headers = new Headers();
+            $headers->addHeaders($user);
+            $request->setHeaders($headers);
+        } else {
+            $request->setServer(new \Laminas\Stdlib\Parameters($user));
+        }
+        return $request;
+    }
+
+    public function testLogin1()
```

Comment:
Would be helpful to have doc comments on the tests explaining their purpose, for future reference.